src/benchmarklib/abstract_table_generator.cpp

```diff
@@ -401,9 +401,23 @@ void AbstractTableGenerator::_create_table_indexes(
   std::cout << "- Creating table indexes done (" << format_duration(metrics.table_index_duration) << ")" << std::endl;
 }
 
-AbstractTableGenerator::IndexesByTable AbstractTableGenerator::_indexes_by_table() const {
-  // Indexes can be specified in a derived concrete class by overriding this function.
-  return {};
+AbstractTableGenerator::IndexesByTable AbstractTableGenerator::_indexes_by_table(std::unordered_map<std::string, BenchmarkTableInfo>& table_info_by_name) const {
+  auto indexes_by_table = std::map<std::string, std::vector<std::vector<std::string>>>{};
+
+  for (const auto& table_info_by_name_pair : table_info_by_name) {
+    const auto& table_name = table_info_by_name_pair.first;
+    const auto& table_info = table_info_by_name_pair.second;
+    const auto table = table_info.table;
```

Comment:
Not necessary.
I was wondering if creating references to shared pointers was a good habit.
Is this the way we want to do it?

Reply:
In case we can't get into any problems (e.g., the caller doing something "bad" to the shared_ptr), we usually pass shared_ptr by reference as we know the caller holds a real copy.